xform-anatomy.rst

```diff
@@ -0,0 +1,420 @@
+
+If you specify a value for :``<mystring>`` in the original XForm, it presents that as the default answer to the prompt:
+
+.. code-block:: xml
+
+  <mystring>Default answer</mystring>
+
+Adding more prompts is as simple as adding more variables:
+
+.. code-block:: xml
+
+  <instance>
+  <data id="myotherform" >
+    <meta>
+      <instanceID/>
+    </meta>
+    <mystring/>
+    <a_number/>
+    <birthday/>
+    <date>2010-06-15</date>
+    <select>a c</select>
+    <favorite_number>7</favorite_number>
+    <whatever_i_want/>
+  </data>
+  </instance>
+
+.. form-bindings:
+
+Bindings
+---------
+
+Since we have two variables, we need two bindings:
+
+.. code-block:: html
+
+  <bind nodeset="/data/meta/instanceID" type="string" 
+           readonly="true()" calculate="concat('uuid:',uuid())" />
+  <bind nodeset="/data/mystring" type="string"/>
+
+The first binding defines how the unique identifier for a filled-in form will be constructed.
+
+The second binding specifies that ``<mystring>`` will hold a string value. Bindings that do not specify a "type" are assumed to be strings. However, ODK Aggregate issues warnings if a type is not specified because ordering of numbers differs from ordering of strings; specifying a "type" ensures that the proper ordering is applied, i.e., "10" is ordered alphabetically less than "2", but, if these are interpreted as numbers, the order is reversed (10 is greater than 2).
+
+Bindings are also where you specify branching, calculations, read-only fields, required fields, input constraints, etc.
+
+Note that the binding references the instance node using an XPath expression. The expression represents the hierarchy inside the ``<instance>`` tags.
+
+.. code-block:: html
+
+  <instance>
+  <data>
+    <mystring/> <!-- referenced as /data/mystring -->
+  </data>
+  </instance>
+
+In this example, mystring is referenced by: :guilabel:`/data/mystring`.
+
+.. form-body:
+
+Body
+-----
+
+The body represents what should be shown to the user. In this example, we have two variables, but the ``<instanceID>`` variable is merely for bookkeeping. Thus, we will only have one prompt displayed to the user for the ``<mystring>`` variable. This is placed in the body element:
+
+.. code-block:: html
+
+  <h:body>
+  <input ref="/data/mystring">
+    <label>This is a string prompt</string>
+    <hint>You don't need a hint, but you can add one</hint>
+  </input>
+  </h:body>
+
+The type of widget/prompt to show the user is specified by the <input> tag. Where to put the data is specified by the ``ref=""`` attribute. The <label> is what will be shown to the user as the prompt header, and the ``<hint>`` is an optional piece of text to display.
```

Comment:
- The type of form widget is specified by the tag name of the element.
- The ref attribute specifies which node the response will be stored in